Sparse CSR CUDA: fix input checks for `addmm` and `mm`

IvanYashchuk opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #66401
* #62180
* #65606
* #64391
* #61536
* #61858
* #61857
* #63948
* #63511
* #63510
* #63509
* __->__ #66485

The errors for incorrectly sized inputs should match the dense variants
of functions.
Moved addmm_out_sparse_csr_dense_cuda from SparseCsrTensorMath.cu and
removed unnecessary device check.

cc @nikitaved @pearu @cpuhrsch @IvanYashchuk

Differential Revision: [D31764036](https://our.internmc.facebook.com/intern/diff/D31764036)

IvanYashchuk committed:
Sparse CSR CUDA: fix input checks for `addmm` and `mm`

The errors for incorrectly sized inputs should match the dense variants
of functions.
Moved addmm_out_sparse_csr_dense_cuda from SparseCsrTensorMath.cu and
removed unnecessary device check.

[ghstack-poisoned]

IvanYashchuk committed:
Update on "Sparse CSR CUDA: fix input checks for `addmm` and `mm`"


The errors for incorrectly sized inputs should match the dense variants
of functions.
Moved addmm_out_sparse_csr_dense_cuda from SparseCsrTensorMath.cu and
removed unnecessary device check.

cc nikitaved pearu cpuhrsch @IvanYashchuk

[ghstack-poisoned]

IvanYashchuk committed:
Update on "Sparse CSR CUDA: fix input checks for `addmm` and `mm`"


The errors for incorrectly sized inputs should match the dense variants
of functions.
Moved addmm_out_sparse_csr_dense_cuda from SparseCsrTensorMath.cu and
removed unnecessary device check.

cc nikitaved pearu cpuhrsch @IvanYashchuk

[ghstack-poisoned]

cpuhrsch reviewed (approved):
Looks fine but take a look at the comment about TORCH_INTERNAL_ASSERT_DEBUG_ONLY

IvanYashchuk committed:
Update on "Sparse CSR CUDA: fix input checks for `addmm` and `mm`"


The errors for incorrectly sized inputs should match the dense variants
of functions.
Moved addmm_out_sparse_csr_dense_cuda from SparseCsrTensorMath.cu and
removed unnecessary device check.

cc nikitaved pearu cpuhrsch @IvanYashchuk

[ghstack-poisoned]

IvanYashchuk committed:
Update on "Sparse CSR CUDA: fix input checks for `addmm` and `mm`"


The errors for incorrectly sized inputs should match the dense variants
of functions.
Moved addmm_out_sparse_csr_dense_cuda from SparseCsrTensorMath.cu and
removed unnecessary device check.

cc nikitaved pearu cpuhrsch @IvanYashchuk

[ghstack-poisoned]

anjali411 commented:
@anjali411 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31764036).
